File: backend/app/data/schemas/appreciation_log.py
from pydantic import BaseModel, Field
from marshmallow import Schema, fields
from enum import Enum

class AppreciationLogCreateReq(BaseModel):
    old_amount: int
    new_amount: int
    customer_investment_id: int

class AppreciationLogResDaily(Schema):
    old_amount = fields.Int()
    new_amount = fields.Int()
    created_at = fields.Str()

class AppreciationLogResWeekly(AppreciationLogResDaily):
    week = fields.Int()
    year = fields.Int()

class AppreciationLogResMonthly(AppreciationLogResDaily):
    month = fields.Int()
    year = fields.Int()

class AppreciationLogResQuarterly(AppreciationLogResDaily):
    quarter = fields.Int()
    year = fields.Int()

class AppreciationLogReqQuery(Enum):
    WEEKLY = "weekly"
    MONTHLY = "monthly"
    QUARTERLY = "quarterly"

File: frontend/src/types/appreciation_log.ts
export interface AppreciationLogRes {
    old_amount: number;
    new_amount: number;
    created_at: string;
}

export interface AppreciationLogResWeekly extends AppreciationLogRes {
    week: number;
    year: number;
}

export interface AppreciationLogResMonthly extends AppreciationLogRes {
    month: number;
    year: number;
}

export interface AppreciationLogResQuarterly extends AppreciationLogRes {
    quarter: number;
    year: number;
}

export enum AppreciationLogType {
    WEEKLY = "weekly",
    MONTHLY = "monthly",
    QUARTERLY = "quarterly"
}